src/ui/components/inputs/EmailInputWithSpellingHelp/useEmailSpellingHelp.native.test.tsx

```diff
@@ -0,0 +1,84 @@
+import { act, renderHook } from 'tests/utils'
+
+import { useEmailSpellingHelp, SUGGESTION_DELAY_IN_MS } from './useEmailSpellingHelp'
+
+const initialProps: Parameters<typeof useEmailSpellingHelp>[0] = {
+  email: '',
+}
+
+jest.useFakeTimers()
+
+describe('useEmailSpellingHelp', () => {
+  it.each(['', undefined])(
+    'should not display suggestion for empty email',
+    (email: string | undefined) => {
+      const { result } = renderHook(useEmailSpellingHelp, {
+        initialProps: { ...initialProps, email },
+      })
+
+      expect(result.current.suggestedEmail).toBeUndefined()
+    }
+  )
+
+  it('should display suggestion with a corrected email when the email is mystyped', async () => {
+    const { result, rerender } = renderHook(useEmailSpellingHelp, { initialProps })
+
+    await act(async () => {
+      rerender({
+        ...initialProps,
+        email: 'firstname.lastname@gmal.com',
+      })
+    })
+
+    await act(async () => {
+      jest.advanceTimersByTime(SUGGESTION_DELAY_IN_MS)
+    })
+
+    expect(result.current.suggestedEmail?.full).toEqual('firstname.lastname@gmail.com')
+  })
+
+  it('should not display suggestion before a delay', async () => {
+    const { result, rerender } = renderHook(useEmailSpellingHelp, { initialProps })
+
+    await act(async () => {
+      rerender({
+        ...initialProps,
+        email: 'firstname.lastname@gmal.com',
+      })
+    })
+
+    await act(async () => {
+      jest.advanceTimersByTime(SUGGESTION_DELAY_IN_MS - 1)
+    })
+
+    expect(result.current.suggestedEmail).toBeUndefined()
+  })
+
+  it('should not display suggestion before a delay with the last typed stuff', async () => {
```

Comment:
J'ai pas très bien compris ce que faisait ce test ? (peut être utiliser email à la place de stuff mais je sais pas si ça le rend vraiment plus clair...)

Reply:
ce test vérifie qu'on utilise bien un `debounce`

mais c'est difficile de le nommer

Reply:
Et c'est quoi la différence avec le test du dessus du coup ? (Je vois qu'il y a une différence mais j'arrive pas à bien la comprendre je pense ahah)

Reply:
le test du dessus, un simple `setTimeout` suffit pour le faire passer

est-ce que tu penses qu'il faudrait supprimer l'un des deux tests ?

Reply:
J'ai l'impression que celui-ci est redondant avec celui d'avant plus ou moins oui 🧐 on pourrait l'appeler "should not display suggestion before a delay once the user finish typing their email' ou quelque chose comme ça ?